notebooks/Fine_tuning_Code_LLM_on_single_GPU.ipynb

```diff
@@ -0,0 +1,5363 @@
+{
```

Comment:
<p>Why is <code>flash-attn</code> needed here?  </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/huggingface/cookbook/pull/8/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Reply:
<p>When we load the model with <code>from_pretrained</code> use it for faster training. I mention it briefly in the paragraph above. Do you think it's worth explaining more? </p><p>Now that you mention it, I notice that I should use <code>attn_implementation="flash_attention_2"</code>   instead of <code>use_flash_attention_2=True</code> which is deprecated.  </p>